amethyst_renderer/src/tex.rs

```diff
@@ -159,7 +160,7 @@ where
         let mut data = self.data.as_ref();
 
         if cfg!(feature = "opengl") {
-            let pixel_width = (self.info.format.get_total_bits() / 8) as usize;
+            let pixel_width = (self.info.format.get_total_bits() / 8) as usize / size_of::<T>();
```

Comment:
Good catch!

Reply:
Yeah it was pretty confusing when my f32 textures were panicking here, I eventually figured out it was because size_of::<f32>() was greater than 1, which hasn't been true for any of our other textures thus far miraculously.